sw/device/tests/flash_ctrl_rma_test.c

```diff
@@ -0,0 +1,213 @@
+    }                                                                    \
+  } while (0)
+
+static status_t write_info_page(dif_flash_ctrl_state_t *flash, uint32_t page_id,
+                                const uint32_t *data, bool scramble) {
+  uint32_t address = 0;
+  if (scramble) {
+    TRY(flash_ctrl_testutils_info_region_scrambled_setup(
+        flash, page_id, kFlashInfoBankId, kFlashInfoPartitionId, &address));
+  } else {
+    TRY(flash_ctrl_testutils_info_region_setup(
+        flash, page_id, kFlashInfoBankId, kFlashInfoPartitionId, &address));
+  }
+
+  TRY(flash_ctrl_testutils_erase_and_write_page(
+      flash, address, kFlashInfoPartitionId, data,
+      kDifFlashCtrlPartitionTypeInfo, 1));
+
+  LOG_INFO("wr_info: data:0x%x", *data);
+  return OK_STATUS();
+}
+
+static status_t read_info_page(dif_flash_ctrl_state_t *flash, uint32_t page_id,
+                               uint32_t *data, bool scramble) {
+  uint32_t address = 0;
+  if (scramble) {
+    TRY(flash_ctrl_testutils_info_region_scrambled_setup(
+        flash, page_id, kFlashInfoBankId, kFlashInfoPartitionId, &address));
+  } else {
+    TRY(flash_ctrl_testutils_info_region_setup(
+        flash, page_id, kFlashInfoBankId, kFlashInfoPartitionId, &address));
+  }
+
+  TRY(flash_ctrl_testutils_read(flash, address, kFlashInfoPartitionId, data,
+                                kDifFlashCtrlPartitionTypeInfo, 1, 1));
+
+  return OK_STATUS();
+}
+
+static status_t read_and_check_info(bool match) {
+  uint32_t readback_data[3];
+  read_info_page(&flash, kFlashInfoPageIdCreatorSecret, readback_data, true);
+  read_info_page(&flash, kFlashInfoPageIdOwnerSecret, readback_data + 1, true);
+  read_info_page(&flash, kFlashInfoPageIdIsoPart, readback_data + 2, true);
+  LOG_INFO("readdata0: %x", *readback_data);
+  LOG_INFO("readdata1: %x", *(readback_data + 1));
+  LOG_INFO("readdata2: %x", *(readback_data + 2));
+  if (match) {
+    // iso partition cannot be accessed in dev state.
+    // check create and owner partition only
+    CHECK_ARRAYS_EQ(readback_data, kRandomData, 2);
+  } else {
+    CHECK_ARRAYS_NE(readback_data, kRandomData, 3);
+  }
+  LOG_INFO("read_and_check is done");
+  return OK_STATUS();
+}
+
+bool sram_main(void) {
+  // Initialize UART console.
+  CHECK_DIF_OK(dif_pinmux_init(
+      mmio_region_from_addr(TOP_EARLGREY_PINMUX_AON_BASE_ADDR), &pinmux));
+  CHECK_DIF_OK(dif_uart_init(
+      mmio_region_from_addr(TOP_EARLGREY_UART0_BASE_ADDR), &uart));
+  pinmux_testutils_init(&pinmux);
+  CHECK_DIF_OK(dif_flash_ctrl_init_state(
+      &flash, mmio_region_from_addr(TOP_EARLGREY_FLASH_CTRL_CORE_BASE_ADDR)));
+  CHECK_DIF_OK(dif_otp_ctrl_init(
+      mmio_region_from_addr(TOP_EARLGREY_OTP_CTRL_CORE_BASE_ADDR), &otp_ctrl));
+
+  CHECK(kUartBaudrate <= UINT32_MAX, "kUartBaudrate must fit in uint32_t");
+  CHECK(kClockFreqPeripheralHz <= UINT32_MAX,
+        "kClockFreqPeripheralHz must fit in uint32_t");
+  CHECK_DIF_OK(dif_uart_configure(
```

Comment:
Looking at the code of `ottf_console_init()`, I think the uart is already initialized there and this should not be necessary

Reply:
> Looking at the code of `ottf_console_init()`, I think the uart is already initialized there and this should not be necessary. The above checks on the baudrate are also checked there.

Good point. removed uart from the test file and BUILD.